actions/views/root.js

```diff
@@ -8,18 +8,22 @@ import {Client4} from 'mattermost-redux/client';
 import {ActionTypes} from 'utils/constants';
 
 export function loadMeAndConfig() {
-    return (dispatch) => {
+    return async (dispatch) => {
         // if any new promise needs to be added please be mindful of the order as it is used in root.jsx for redirection
         const promises = [
             dispatch(getClientConfig()),
             dispatch(getLicenseConfig()),
         ];
 
+        // need to await for clientConfig first as it is required for loadMe
+        const resolvedPromises = await Promise.all(promises);
+
+        let loadMeResponse;
         if (document.cookie.indexOf('MMUSERID=') > -1) {
-            promises.push(dispatch(UserActions.loadMe()));
+            loadMeResponse = await dispatch(UserActions.loadMe());
         }
 
-        return Promise.all(promises);
+        return [...resolvedPromises, loadMeResponse];
```

Comment:
If there is no `MMUSERID` cookie, `loadMeResponse` will be `undefined`, whereas previously it wasn't returned at all. Since we're creating a new array on line 26 anyway, I'd just drop the `const` on `resolvedPromises` and `push` as before.

Reply:
@lieut-data Good catch. Updated.